pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1585,18 +1595,34 @@ func (r *ReconcileInfinispan) serviceExternal(m *infinispanv1.Infinispan) *corev
 			Namespace: m.ObjectMeta.Namespace,
 			Labels:    lsService,
 		},
-		Spec: corev1.ServiceSpec{
+	}
+
+	if externalServiceType == corev1.ServiceTypeLoadBalancer {
+		externalService.Spec = corev1.ServiceSpec{
+			Type:     externalServiceType,
+			Selector: lsPodSelector,
+		}
+	} else if externalServiceType == corev1.ServiceTypeNodePort {
+		var portNum int32
+		if m.Spec.Expose.Ports != nil || len(m.Spec.Expose.Ports) > 0 {
+			portNum = m.Spec.Expose.Ports[0].NodePort
+		} else {
+			portNum = 0
+		}
+		externalService.Spec = corev1.ServiceSpec{
 			Type:     externalServiceType,
 			Selector: lsPodSelector,
 			Ports: []corev1.ServicePort{
 				{
 					Port:       int32(11222),
 					TargetPort: intstr.FromInt(11222),
-					// Fix NodePort to match exported port in kind configuration
-					NodePort: int32(30222),
+					NodePort:   portNum,
```

Comment:
Have you considered not defining the NodePort at all? This part will be added automatically by OpenShift if the service type will be NodePort. This would allow you to have the definition same way as before and would have save a lot of code lines.

Reply:
If the node port specification would be required feature present in Infinispan CR then you could just check that the service is of type NodePort with node port set explicitly and add it extra. Also this would allow you too keep the support for all the service types as it was.

Reply:
And if the NodePort should be part of the Infinispan CR then design doc should be updated.

Reply:
I've reorganized the switch code.
The `.Expose` field is a v1.ServiceSpec so it implicitly contains NodePort info.

tbh I would leave to the user the task to expose things in the k8s proper way. atm we just save the user from the effort to write 11222 two times